includes/widgets/class-amp-widget-categories.php

```diff
@@ -0,0 +1,102 @@
+<?php
+/**
+ * Class AMP_Widget_Categories
+ *
+ * @package AMP
+ */
+
+/**
+ * Class AMP_Widget_Categories
+ *
+ * @package AMP
+ */
+class AMP_Widget_Categories extends WP_Widget_Categories {
+
+	/**
+	 * Adds an 'on' attribute to the category dropdown's <select>.
+	 *
+	 * @param string $dropdown The markup of the category dropdown.
+	 * @return string $dropdown The filtered markup of the dropdown.
+	 */
+	public function modify_select( $dropdown ) {
+		$new_select = sprintf( '<select on="change:widget-categories-dropdown-%d.submit"', esc_attr( $this->number ) );
+		return str_replace( '<select', $new_select, $dropdown );
+	}
+
+	/**
+	 * Echoes the markup of the widget.
+	 *
+	 * Mainly copied from WP_Widget_Categories::widget()
+	 * There's now an id for the <form>.
+	 * And the dropdown is now filtered with 'wp_dropdown_cats.'
+	 * This enables adding an 'on' attribute, with the id of the form.
+	 * So changing the dropdown value will redirect to the category page, with valid AMP.
+	 *
+	 * @param array $args Widget display data.
+	 * @param array $instance Data for widget.
+	 * @return void.
+	 */
+	public function widget( $args, $instance ) {
+		static $first_dropdown = true;
+		// @codingStandardsIgnoreLine
```

Comment:
What is this for? Is it for the text domain? If so, I suggest editing the `phpcs.xml` to include `default`, such as:

```xml
<property name="text_domain" value="amp,default" />
```

And then add `'default'` as the text domain for `__( 'Categories' )` and others.

Reply:
Thanks, @westonruter. That `@codingStandardsIgnoreLine` was for the text domain. Thanks for suggesting a solution. It's [applied here](https://github.com/Automattic/amp-wp/pull/870/files#diff-6b216d983cf0a62e9c648934c5687456R10).